x/did/types/document_sign.go

```diff
@@ -0,0 +1,152 @@
+	ariesdid "github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/ld"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/jsonld"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/signer"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/suite"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/suite/ecdsasecp256k1signature2019"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/verifier"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/context"
+	ldstore "github.com/hyperledger/aries-framework-go/pkg/store/ld"
+	jsld "github.com/piprate/json-gold/ld"
+)
+
+func SignDocument(doc []byte, verificationID string, sequence uint64, privKey *btcec.PrivateKey) ([]byte, error) {
+
+	signerContext := NewSignerContext(verificationID, ES256K_2019_SIG, sequence)
+	documentSigner := GetECDSASigner(privKey)
+	loader, err := GetDocumentLoader()
+	if err != nil {
+		return nil, fmt.Errorf("error to get document loader: %v", err)
+	}
+
+	signedDocument, err := documentSigner.Sign(signerContext, doc, jsonld.WithDocumentLoader(loader))
+	if err != nil {
+		return nil, fmt.Errorf("error to sign document: %v", err)
+	}
+
+	return signedDocument, nil
+}
+
+func VerifyProof(doc ariesdid.Doc) error {
+
+	sigVerifier := ecdsasecp256k1signature2019.New(suite.WithVerifier(ecdsasecp256k1signature2019.NewPublicKeyVerifier()))
+	loader, err := GetDocumentLoader()
+	if err != nil {
+		return err
+	}
+	if err := doc.VerifyProof([]verifier.SignatureSuite{sigVerifier}, jsonld.WithDocumentLoader(loader)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func GetDocumentLoader() (*ld.DocumentLoader, error) {
+	storeProvider := mem.NewProvider()
+	contextStore, err := ldstore.NewContextStore(storeProvider)
+	if err != nil {
+		return nil, err
+	}
+
+	remoteProviderStore, err := ldstore.NewRemoteProviderStore(storeProvider)
+	if err != nil {
+		return nil, err
+	}
+	ctx, err := context.New(
+		context.WithJSONLDContextStore(contextStore),
+		context.WithJSONLDRemoteProviderStore(remoteProviderStore),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	loader, err := ld.NewDocumentLoader(
+		ctx,
+		ld.WithRemoteDocumentLoader(&httpRemoteDocumentLoader{}),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return loader, nil
+}
+
+type httpRemoteDocumentLoader struct{}
+
+func (l *httpRemoteDocumentLoader) LoadDocument(url string) (*jsld.RemoteDocument, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	doc, err := jsld.DocumentFromReader(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return &jsld.RemoteDocument{
+		DocumentURL: url,
+		Document:    doc,
+	}, nil
+}
+
+func NewSignerContext(vmID string, signatureType string, sequence uint64) *signer.Context {
+	return &signer.Context{
+		SignatureType: signatureType,
+		Creator:       vmID,
+		Domain:        strconv.FormatUint(sequence, 10),
```

Comment:
It'd be clearer to use `Nonce` over `Domain`. You can convert uint64 into []byte: https://go.dev/play/p/BN_4jwLXilk. 

The `Domain` is usually for the verifier to specify the scope in which the `Nonce` has a meaning.

Reply:
As we've discussed in other channels, `Nonce` don't have any effect on the generation of proof values, so they can't be used as a defense against replay-attacks. I will look at the role of `Domain` and `Nonce` in detail, but for now, let's leave it at that.